compiler/src/main/scala/org/scalajs/nscplugin/PrepJSInterop.scala

```diff
@@ -641,17 +641,17 @@ abstract class PrepJSInterop[G <: Global with Singleton](val global: G)
           membSym.defStringSeenAs(sym.thisType.memberType(membSym))
 
         // Check for overrides with different JS names - issue #1983
-        if (jsInterop.jsNameOf(low) != jsInterop.jsNameOf(high)) {
+        if (jsInterop.JSCallingConvention.of(low) != jsInterop.JSCallingConvention.of(high)) {
```

Comment:
This change probably stretches our notion of source v.s. binary compatibility. IIUC, this could now fail at usage site of a library compiled with a older version of Scala.js.

Unsure though what to do about this (IIUC detecting if the current class is "the problem" is hard).